### PR TITLE
Remove clippy warnings from massa-pool

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

File: massa-pool/src/error.rs [hunk 1]
```diff
@@ -11,11 +11,17 @@ pub enum PoolError {
...
     JoinError(#[from] tokio::task::JoinError),
     /// models error: {0}
     ModelsError(#[from] ModelsError),
 }
+
+impl From<ProtocolError> for PoolError {
+    fn from(protocol_error: ProtocolError) -> Self {
+        PoolError::ProtocolError(Box::new(protocol_error))
+    }
+}
```
Comment (AureliaDolo) [inline massa-pool/src/error.rs:27]: What's the purpose of that ?

**Review by AureliaDolo — CHANGES_REQUESTED** [review]
Cargo fmt !

File: massa-pool/src/error.rs (hunk 1 above)
Comment (yvan-sraka) [reply massa-pool/src/error.rs:27]: I guess helping implicit error conversion (the power of `?` operator and `anyhow::Error` which is behind the hood a `Box<dyn std::error::Error>`)!

**Review by AureliaDolo — APPROVED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Comment by adrien-zinger** [comment]
bors merge

**Pull request merged**